### PR TITLE
Simplify seccomp logic

### PR DESCRIPTION
**Pull request by pjbgf**
Simplifies the existent seccomp code without impacting behaviour. 

This is a follow-up PR from https://github.com/moby/moby/pull/41320, which adds test coverage.

image
img source: https://www.dogbreedinfo.com/aussiedoodle.htm

Signed-off-by: Paulo Gomes <pjbgf@linux.com>

### CODE REVIEW

File: daemon/seccomp_linux.go [hunk 1]
```diff
@@ -22,18 +22,18 @@ func WithSeccomp(daemon *Daemon, c *container.Container) coci.SpecOpts {
...
+			if c.SeccompProfile != "" {
 				return fmt.Errorf("seccomp is not enabled in your kernel, cannot run a custom seccomp profile")
 			}
 			logrus.Warn("seccomp is not enabled in your kernel, running container without default profile")
 			c.SeccompProfile = "unconfined"
-		}
-		if c.SeccompProfile == "unconfined" {
 			return nil
 		}
 		if c.SeccompProfile != "" {
```
Comment (thaJeztah) [inline daemon/seccomp_linux.go:39]: Perhaps these nested ifs could be rewritten to a `switch`. Combined with my other comment, something like;

```go
// WithSeccomp sets the seccomp profile
func WithSeccomp(daemon *Daemon, c *container.Container) coci.SpecOpts {
	return func(ctx context.Context, _ coci.Client, _ *containers.Container, s *coci.Spec) error {
		if c.SeccompProfile == "unconfined" {
			return nil
		}
		if c.HostConfig.Privileged {
			return nil
		}
		if !daemon.seccompEnabled {
			if c.SeccompProfile != "" {
				return fmt.Errorf("seccomp is not enabled in your kernel, cannot run a custom seccomp profile")
			}
			logrus.Warn("seccomp is not enabled in your kernel, running container without default profile")
			c.SeccompProfile = "unconfined"
			return nil
		}

		var err error
		switch {
		case c.SeccompProfile != "":
			s.Linux.Seccomp, err = seccomp.LoadProfile(c.SeccompProfile, s)
		case daemon.seccompProfile != nil:
			s.Linux.Seccomp, err = seccomp.LoadProfile(string(daemon.seccompProfile), s)
		default:
			s.Linux.Seccomp, err = seccomp.GetDefaultProfile(s)
		}
		return err
	}
}
```

**Review by thaJeztah — CHANGES_REQUESTED** [review]
thanks! left some suggestions (feedback welcome 👍)

File: daemon/seccomp_linux.go (hunk 1 above)
Comment (pjbgf) [reply daemon/seccomp_linux.go]: That's awesome, the final result does look a lot cleaner. Thank you. 👍

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@cpuguy83 ptal

**Comment by thaJeztah** [comment]
@akihirosuda @cpuguy83 PTAL

**Review by tiborvass — APPROVED** [review]

**Pull request merged**